filecoin-proofs/src/bin/phase2.rs

```diff
@@ -1423,6 +1496,74 @@ fn main() {
 
                 convert_small(path_before)
             }
+            "split-keys" => {
+                let path_input = matches.value_of("path-input").unwrap();
+
+                println!("reading params: {}", path_input);
+
+                // Get the identifier for the output files based in the input file's name
+                let (proof, _hasher, sector_size_enum, _head, param_num, param_size, _read_raw) =
+                    parse_params_filename(path_input);
+                assert!(param_size.is_large(), "params must be large");
+                let sector_size = sector_size_enum.as_u64();
+                let identifier = with_shape!(sector_size, parameter_identifier, sector_size, proof);
+
+                let mut input_file = File::open(path_input).unwrap();
+                // Get the vk data from the input file and write it out again. It's not that large,
+                // so the penalty of deserializing and serializing again should be negligible.
+                {
+                    let vk_data = groth16::VerifyingKey::<Bls12>::read(&input_file)
+                        .expect("failed to deserialize vk from input file");
+                    let vk_path = verifying_key_id(&identifier);
+                    println!("writing verifying key to file: {}", vk_path);
+                    let mut vk_file = File::create(&vk_path).unwrap();
+                    vk_data.write(&mut vk_file).unwrap();
+                }
+
+                let vk_file_size = input_file.seek(SeekFrom::Current(0)).unwrap();
+                println!("size of the verifying key is {} bytes", vk_file_size);
+
+                {
+                    let params_path = parameter_id(&identifier);
+                    println!("writing parameters to file: {}", params_path);
+                    let mut params_file = File::create(&params_path).unwrap();
+
+                    // input_file_size - vk_data - cs_hash - contributions_length -
+                    //   (num_contributions * public_key_size)
+                    let params_file_size = input_file.metadata().unwrap().len()
+                        - vk_file_size
+                        - 64
+                        - 4
+                        - (param_num as u64 * 544);
+                    println!("size of the parameters file is {} bytes", params_file_size);
+                    io::copy(
+                        &mut Read::by_ref(&mut input_file).take(params_file_size),
+                        &mut params_file,
+                    )
+                    .unwrap();
+                }
+
+                {
+                    // Writing the contributions to disk is not needed for the final parameters,
+                    // they won't be published, they are only there for verification purpose.
+                    let contribs_path =
+                        format!("v{}-{}.contribs", parameter_cache::VERSION, &identifier);
```

Comment:
There is a subtlety here. If we use the current value of `VERSION` when running the tool, then we will end up with the version corresponding to the current (untrusted) parameters.

That would imply a sequence like the following:

- Bump `VERSION`.
- Run split tool to create trusted parameters.
- Publish parameters.
- Release.

Note that in this sequence, the version must be bumped before the tool is run. We *could* bump the version here/now, but then the trusted parameters would need to be included in the very next release. For that reason, something like the sequence above is probably better. It's also possible that I'm seeing something wrong, so please let me know if so or explicitly confirm below otherwise (cc: @cryptonemo @vmx ).

Reply:
I would use the sequence then @cryptonemo would use when he publishes new parameters. I'd expect it to be what @porcuquine outlined above, but @cryptonemo would know best.

Reply:
@cryptonemo Can you confirm that we're not missing anything here and that my sequence above is workable (or propose an alternative)?

Reply:
Confirmed